apps/src/p5lab/spritelab/commands/spriteCommands.js

```diff
@@ -40,10 +40,14 @@ export const commands = {
   getThisSprite(which, extraArgs) {
     if (extraArgs) {
       if (which === 'this') {
-        return {id: extraArgs.sprite};
+        if (extraArgs.clickedSprite !== undefined) {
+          return {id: extraArgs.clickedSprite};
+        } else if (extraArgs.subjectSprite !== undefined) {
+          return {id: extraArgs.subjectSprite};
+        }
```

Comment:
NIT: (up to you) should there be an `else {throw / console.warn()}` here?

Reply:
No, for generated code we like to fail gracefully/silently whenever possible. The `else` case here would occur any time someone uses a `this/other sprite` block outside of an event. While incorrect for _their_ code, that's certainly not the type of error/event that we care about tracking.